app/src/main/cpp/asurfacerenderer/asurface_jni.cpp

```
@@ -0,0 +1,181 @@
+}
+
+extern "C" JNIEXPORT void JNICALL
+Java_com_winlator_renderer_ASurfaceRenderer_nativeScanoutSetCursorImage(
+        JNIEnv* env, jobject, jobject buf, jshort w, jshort h, jshort stride)
+{
+    if (!buf) return;
+    void* px = env->GetDirectBufferAddress(buf);
+    if (px && env->GetDirectBufferCapacity(buf) >= (jlong)w*h*4) {
+        std::shared_lock lk(g_ctxMutex);
+        if (auto* r = g_ctx) r->scanoutSetCursorImage(px, w, h, stride);
+    }
```

Cppcheck (2.21.0)

[error] Found an exit path from function with non-void return type that has missing return statement

(missingReturn)